pkg/pushsync/pushsync.go

```diff
@@ -127,7 +127,7 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 	defer cancel()
 	defer func() {
 		if err != nil {
-			ps.metrics.TotalErrors.Inc()
+			ps.metrics.TotalHandlerErrors.Inc()
```

Comment:
Should TotalHandlerErrors be added alongside TotalErrors as this error should be counted in to TotalErrors as well?

Reply:
so we were going through this yesterday while creating KPIs for pushsync and we wanted to separate success rates for different functions. the problem is that TotalErrors if added here would be potentially incremented twice per forwarding operation, which means we cannot use it to build a success rate of a certain code section